MDC: Adding 2024-08-01-preview version to SecurityConnectors API [Introducing new JFrog Environment]

Norfolc opened the pull request:
This Pull Request already reviewed in the private repo: Azure/azure-rest-api-specs-pr#19861

# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

rkmanda commented on specification/security/resource-manager/Microsoft.Security/preview/2024-08-01-preview/securityConnectors.json:
```diff
@@ -0,0 +1,1785 @@
+{
+  "swagger": "2.0",
```
>"swagger": "2.0", [](http://example.com/codeflow?start=2&length=17)

[ARMBlockingComment] All types from previous versions need to be carried forward to the new API version

Norfolc replied:
can you please elaborate on this? all types from this file are carried to the new API version.

gary-x-li replied:
Confirmed with @rkmanda that the original comment no longer applies since you've reset the commits so we can see the proper diffs for this PR now.

rkmanda commented on specification/security/resource-manager/Microsoft.Security/preview/2024-08-01-preview/securityConnectors.json:
```diff
@@ -0,0 +1,1785 @@
+                "modelAsString": true
+              }
+            },
+            "exclusionTags": {
```
>       "exclusionTags": { [](http://example.com/codeflow?start=5&length=25)

Hard to tell whats changing.  The first commit needs to be an exact copy of the previous API version. All new changes should only be added in the subsequent commits. 
 This allows the reviewer to get a clear understanding of the actual changes being introduced. With the way the PR is raised now, it is not possible for the reviewer to tell what the changes are. Please either abandon the PR and raise another one with the recommendation or create a new set of commits on this PR following the recommendation. If you are doing the pattern please indicate which commit is the exact copy of the previous version.
 
 [https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version) . Please remove the "ARMChangesRequested" label once you have fixed this issue.

Norfolc replied:
I followed the instructions from [Move private specs to public](https://aka.ms/azsdk/move-pr).
The PR was approved in private repo: Azure/azure-rest-api-specs-pr#19861

Norfolc replied:
Created a new set of commits as you requested.

Norfolc committed:
Copy files from preview/2024-07-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

Norfolc committed:
Update version to preview/2024-08-01-preview

Updated the API version from preview/2024-07-01-preview to preview/2024-08-01-preview.

Norfolc committed:
Added JFrog Artifactory connector

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[common](https://apiview.dev/Assemblies/Review/da811d14b94f40769492b2ede877e48b?revisionId=f0f79445c8ba4befae2303d787b1f290)
